mysql-test/suite/rpl/t/bug85158.test

```diff
@@ -0,0 +1,109 @@
+#
+# Testcase for bug 85158.
+#
+--source include/not_group_replication_plugin.inc
+--source include/have_gtid.inc
+--source include/have_perfschema.inc
+--source include/have_binlog_format_row.inc
+
+--let $rpl_multi_source= 1
+--let $rpl_topology= 1->2,2->3,1->3
+--source include/rpl_init.inc
+
+# MI writes threshold, if the difference between the count of writes to
+# the MI repository before and after the actions is greater than the
+# threshold, the test failed
+--let $threshold=5
+
+# The number of inserts on server_1, if the bug is not fixed the amount
+# of the MI repository writes will be greater or equal to this value
+--let $inserts_number=50
+
+--echo ###
+--echo # STOP SLAVE ON 2
+--echo ######
+--let $rpl_connection_name= server_2
+--source include/rpl_connection.inc
+STOP SLAVE;
+
+--echo ###
+--echo # CREATE TABLE AND INSERTS ON 1
+--echo ######
+--let $rpl_connection_name= server_1
+--source include/rpl_connection.inc
+CREATE TABLE t1(f1 INT, f2 BLOB);
+--disable_query_log
+--let $j=$inserts_number
+while($j)
+{
+--eval INSERT INTO t1(f1) VALUES($j)
+--dec $j
+}
+--enable_query_log
+
+--echo ###
+--echo # SYNC 1->3
+--echo ######
+--let $sync_slave_connection= server_3
+--source include/sync_slave_sql_with_master.inc
+
+--echo ###
+--echo # MEASURE 3
+--echo ######
+STOP SLAVE;
+UPDATE performance_schema.setup_objects
+  SET enabled='yes', timed='yes'
+  WHERE object_schema='mysql';
+START SLAVE;
+
+let $before=
+`SELECT count_write
+  FROM performance_schema.table_io_waits_summary_by_table
+  WHERE object_schema='mysql' AND object_name ='slave_master_info'`;
+
+--echo ###
+--echo # START SLAVES 2
+--echo ######
+--let $rpl_connection_name= server_2
+--source include/rpl_connection.inc
+START SLAVE;
+
+--echo ###
+--echo # SYNC 1->2,2->3
+--echo ######
+--let $rpl_connection_name= server_1
+--source include/rpl_connection.inc
+--let $sync_slave_connection= server_2
+--source include/sync_slave_sql_with_master.inc
+--let $sync_slave_connection= server_3
+--source include/sync_slave_sql_with_master.inc
+
+--echo ###
+--echo # MEASURE 3
+--echo ######
+let $after=
+`SELECT count_write
+  FROM performance_schema.table_io_waits_summary_by_table
+  WHERE object_schema='mysql' AND object_name ='slave_master_info'`;
+
+--echo ###
+--echo # COMPARE THE MEASUREMENT RESULTS
+--echo ######
+--let $result=`SELECT (($after - $before) < $threshold)`
+--let $assert_text= No more than $threshold writes should have happened to slave_master_info
+--let $assert_cond= $result = 1
+--source include/assert.inc
+
+--echo ###
+--echo # CLEANUP
+--echo ######
+UPDATE performance_schema.setup_objects
+  SET enabled='no', timed='no'
+  WHERE object_schema='mysql';
+--let $rpl_connection_name= server_1
+--source include/rpl_connection.inc
+DROP TABLE t1;
+--disable_query_log
+--let $rpl_skip_sync= 1
```

Comment:
An old question still stands: why is sync skipped here? Based on the previous answers, it must be somehow related to MTR framework wiping the datadirs, but what's the relation?

Reply:
My previous answer is based on mtr source code, see above for details: "So there is no need to restore server state if it has custom config."

1) So, the test is already finished, we got all results we wanted to get, there is no need to spend time to sync servers if they are restarted right just after the test is finished.

2) see rpl_sync.inc:
if ($rpl_multi_source)
{
  --die ERROR IN TEST: rpl_sync.inc cannot be used when $rpl_multi_source=1. If you use multi-source, you must manually sync all servers using sync_slave_sql_with_master.inc or equivalent.
}

Reply:
I disagree with 1. - the servers syncing shows that the replication state is not corrupted by the previous test actions.  But here it looks like 2. would require adding manual syncs and still require rpl_skip_sync for rpl_end, so probably OK to leave as is